core/lib/services/writables.spec.ts

```diff
@@ -0,0 +1,47 @@
+import {writable} from '@amadeus-it-group/tansu';
+import type {SpyInstance} from 'vitest';
+import {beforeEach, describe, expect, test, vi} from 'vitest';
+import {writableWithDefault} from './stores';
+import {typeArray} from './writables';
+
+describe(`Writables service`, () => {
+	let consoleErrorSpy: SpyInstance<Parameters<typeof console.error>, ReturnType<typeof console.error>>;
+
+	beforeEach(() => {
+		consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+	});
+
+	test(`should support typeArray`, () => {
+		const config$ = writable([0]);
+		const store$ = writableWithDefault<number[]>([0], config$, typeArray);
+		// set the wrong value
+		store$.set(100 as any);
+		expect(consoleErrorSpy).toHaveBeenCalledWith('Not setting invalid value', 100);
+		// the store is not updated with the wrong value
+		expect(store$()).toStrictEqual([0]);
+
+		const testArray = [15, 20];
+		store$.set(testArray);
+
+		expect(store$()).toStrictEqual([15, 20]);
+
+		testArray[1] = 15;
+		store$.set(testArray);
+
+		expect(store$()).toStrictEqual([15, 15]);
+	});
+
+	test(`typeArray equal function should compare values`, () => {
```

Comment:
To fix typings, you can use `typeArray.equal!` instead of  `typeArray.equal` in all those lines (or do `const equal = typeArray.equal!` and use `equal` in the following lines).